src/formatDate.js

```diff
@@ -8,7 +8,23 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const newData = date.split(separator);
+  const newFormat = Array.from(toFormat);
+  const newSeparator = newFormat.pop();
+  const obj = {};
+
+  for (let i = 0; i < newData.length; i++) {
+    obj[fromFormat[i]] = newData[i];
+
+    if (fromFormat[i] === 'YY') {
+      obj.YYYY = newData[i] >= 30 ? '19' + newData[i] : '20' + newData[i];
+    } else if (fromFormat[i] === 'YYYY') {
+      obj.YY = newData[i].substring(2);
+    }
+  }
+
+  return newFormat.map((part) => obj[part]).join(newSeparator);
```

Comment:
The `map` function here might produce `undefined` values if the `toFormat` array contains elements that are not keys in the `obj`. It's assumed that `toFormat` will always be correct, but be cautious about potential mismatches.